unblob/handlers/archive/tar.py

```diff
@@ -85,13 +86,8 @@ def _find_end_of_padding(file, *, find_from: int) -> int:
 
 class TarExtractor(Extractor):
     def extract(self, inpath: Path, outdir: Path):
-        tf = SafeTarFile.open(inpath.as_posix())
-        try:
-            tf.extractall(outdir.as_posix())
-        except FileExistsError as file_exists_error:
-            logger.warning(
-                "FileExistsError during tar archive extraction", error=file_exists_error
-            )
+        with contextlib.closing(SafeTarFile(inpath)) as tarfile:
+            tarfile.extractall(outdir)
```

Comment:
Might as well implement `__enter__` and `__exit__` in `SafeTarFile`, no ?

Reply:
yes, they are the same, but going with `__enter__` and `__exit__` is more verbose, and I always forgot the order of the 3 argument names for `__exit__`. It is not fun looking them up and not using them at all.

`contextlib.closing` was the most concise solution, and is relatively well known, I think.